src/vs/workbench/browser/parts/editor/media/tabstitle.css

```diff
@@ -186,10 +194,6 @@
 
 /* No Tab Close Button */
 
-.monaco-workbench > .part.editor > .content > .one-editor-silo > .container > .title .tabs-container > .tab.close-button-off {
```

Comment:
@Dari-K is that an oversight that you removed this rule for good?

Reply:
@bpasero Well the padding-right was moved to another selector so this rule was empty, should I have left the selector with an empty block?

Reply:
@Dari-K but I am no longer seeing the padding when tab buttons are off?

Reply:
@bpasero yeah, you're totally right, and that would make sense, will fix later.

Reply:
@Dari-K ok thanks

Reply:
@bpasero so I fixed the problem with the missing padding, but realised that this solution still may not be good enough. The way it works is moving the padding-right to the label text, which collapses, so when the text isn't overflowing the fade is over the empty padding, and when the padding collapses the fade will be over the text. However the padding-right previously didn't collapse when the tab shrank, so this is a change in behaviour.

Do we have a way to tell when the label is overflowing? Or can you think of another way to hide the fade when we shouldn't be showing it?

Reply:
@Dari-K no we do not, but maybe we could just do this: it seems to work well when tab-close button is on the right and the reason for that is that we reserve some space on the right hand side of the tab for the close button even if the close button is not visible:

![image](https://user-images.githubusercontent.com/900690/37638442-3bef890a-2c0d-11e8-9a42-ae3f2846dcd0.png)

What if we would reserve some space to the right also when tab-close button is left or off in case tab sizing is `shrink`? I assume we would not need much more than `5px`

Reply:
@bpasero I added a pseudo element to reserve 5px on right for sizing shrink with close button left or off like you said. Seems to work well. 👍